platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/GossipEvent.java

```diff
@@ -116,6 +117,14 @@ public Instant getTimeReceived() {
         return timeReceived;
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public long getGeneration() {
+        return hashedData.getGeneration();
```

Comment:
Why change it from accessing the descriptor? Calling this method causes the generation to be calculated every time.

Reply:
Sometimes we need to get the generation of an event before we generate a descriptor for it.

I'm not a big fan of how we currently build descriptors for events. I understand why we do it... we can only generate a descriptor after the event has been hashed. But it adds some really funky restrictions as to which methods can be called at what points in an event's lifecycle.

In this scenario, computing the generation is super cheap (take the max of two numbers and add 1), so I'm not sure caching this value is currently worth the complexity.